language/move-prover/bytecode/src/stackifier_topological_sort.rs

```diff
@@ -0,0 +1,168 @@
+// Copyright (c) The Diem Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::stackless_control_flow_graph::BlockId;
+use std::collections::hash_map::Entry;
+use std::collections::{HashMap, HashSet};
+
+#[derive(Clone, Debug)]
+pub struct StackifierDependency {
+    pub num_prec: usize,
+    pub succ_set: HashSet<BlockId>,
+}
+
+#[derive(Clone, Debug)]
+pub struct StackifierTopologicalSort {
+    pub last_popped_block: Option<BlockId>,
+    top: HashMap<BlockId, StackifierDependency>,
+    // complete topology will not change while popping
+    complete_top: HashMap<BlockId, StackifierDependency>,
+}
+
+impl StackifierDependency {
+    pub fn new() -> StackifierDependency {
+        StackifierDependency {
+            num_prec: 0,
+            succ_set: HashSet::new(),
```

Comment:
Could using a HashSet create some non-deterministic output? We typically use BTreeSet and BTreeMap. HashSet may also work if we are sure it does not influence the order in output.